crates/lakekeeper/src/catalog/tables.rs

```diff
@@ -886,6 +886,18 @@ impl<C: Catalog, A: Authorizer + Clone, S: SecretStore>
             include_deleted: false,
             include_active: true,
         };
+
+        authorized_table_ident_to_id::<C, _>(
+            authorizer.clone(),
+            &request_metadata,
+            warehouse_id,
+            source,
+            list_flags,
+            CatalogTableAction::CanDrop,
+            t.transaction(),
+        )
+        .await?;
```

Comment:
We don't need this here.
`CanDrop` is less specific than `CanRename`, which is checked below. `CanRename` might be implemented as `CanDrop` AND Something else, depending on the Authz system. In our case, for the OpenFGA Authorizer, they are even identical. In any case, the AuthZ system must make sure that `Rename` is as least as restrictive as drop.